docs: reboot CONTRIBUTING.md and PR template to reflect 4-stage manifesto (#18708)
LedgerHQ/ledger-live: .github/pull_request_template.md

<!--
Read CONTRIBUTING.md before opening this PR.

> PRs that skip stages will be sent back to Draft.

Reviewer: **suggestion: Let's phrase this as a short description of the target workflow**

```suggestion
**Create all pull requests in draft**
Automated checks must pass before you make your pull request as "Ready for review". 

See [CONTRIBUTING.md](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md) for guidelines.
```

-->

### 📝 Description

_A clear and concise description of what this PR does and why it is needed._

> _For bug fixes: describe the previous behaviour, the fix, and the non-regression test that would have failed before it._

Reviewer: **suggestion: improve readability**

```suggestion
_For bug fixes: describe the previous behaviour, the fix and the automated check that prevents future regression._
```

_For features: describe the problem solved and the approach taken._
_For libraries: include a usage code sample._

> _Attach screenshots or recordings for visual changes._

Reviewer: **suggestion: let's make the purpose clear**

```suggestion
_When possible, attach screenshots or recordings to help reviewers validate the changes._
```

### 🔗 Context

- **JIRA / GitHub issue**: <!-- e.g. [LLD-1234] or #456 -->
- **ADR** (if any): <!-- link to the relevant architectural decision record -->

> > Before opening: make sure you've gone through the [PR lifecycle stages](../CONTRIBUTING.md#the-pr-lifecycle) — CI green, SonarQube addressed, Copilot reviewed — before converting from Draft to Open.

Reviewer: **suggestion: avoid duplicate messaging**

We have similar information on the first lines of the template

```suggestion
```
